packages/sdk/app-framework/src/plugins/PluginHost/PluginHost.tsx

```diff
@@ -63,65 +66,100 @@ export const PluginHost = ({
       plugins: state.values,
       context: ({ children }) => <PluginProvider value={state.values}>{children}</PluginProvider>,
       root: () => {
-        useEffect(() => {
-          log('initializing plugins', { enabled: state.values.enabled });
-          const timeout = setTimeout(async () => {
-            const enabledIds = [...core, ...state.values.enabled].sort((a, b) => {
-              const indexA = order.findIndex(({ id }) => id === a);
-              const indexB = order.findIndex(({ id }) => id === b);
-              return indexA - indexB;
-            });
-
-            const enabled = await Promise.all(
-              enabledIds
-                .map((id) => definitions[id])
-                // If local storage indicates a plugin is enabled, but it is not available, ignore it.
-                .filter((definition): definition is () => Promise<PluginDefinition> => Boolean(definition))
-                .map((definition) => definition()),
-            );
-
-            const plugins = await Promise.all(
-              enabled.map(async (definition) => {
-                const plugin = await initializePlugin(definition).catch((err) => {
-                  console.error('Failed to initialize plugin:', definition.meta.id, err);
-                  return undefined;
-                });
-                return plugin;
-              }),
-            ).then((plugins) => plugins.filter((plugin): plugin is Plugin => Boolean(plugin)));
-            log('plugins initialized', { plugins });
-
-            await Promise.all(
-              enabled.map(async (pluginDefinition) => {
-                await pluginDefinition.ready?.(plugins);
-              }),
-            );
-            log('plugins ready', { plugins });
-
-            state.values.plugins = shallow(plugins);
-            state.values.ready = true;
-          });
-
-          return () => {
-            clearTimeout(timeout);
-            state.values.ready = false;
-            // TODO(wittjosiah): Does this ever need to be called prior to having dynamic plugins?
-            // void Promise.all(enabled.map((definition) => definition.unload?.()));
-          };
-        }, []);
-
-        if (!state.values.ready) {
-          return <>{fallback}</>;
-        }
-
-        const ComposedContext = composeContext(state.values.plugins);
-
-        return <ComposedContext>{rootComponents(state.values.plugins)}</ComposedContext>;
+        return (
+          <ErrorBoundary fallback={fallback}>
+            <Root order={order} core={core} definitions={definitions} state={state.values} placeholder={placeholder} />
+          </ErrorBoundary>
+        );
       },
     },
   };
 };
 
+const DefaultFallback = ({ error }: { error: Error }) => {
+  return (
+    <div style={{ padding: '1rem' }}>
+      {/* TODO(wittjosiah): Link to docs. */}
+      <p style={{ fontSize: '0.8rem', fontStyle: 'italic' }}>
+        This is the default ErrorBoundary fallback. Consider providing your own!
```

Comment:
Not a fan of conversational defaults. Instead provide an informative and sensible default and document how to change it in the docs.